controllers/front/cronsyncorderfromgateway.php

```diff
@@ -0,0 +1,306 @@
+            $msg .= "Total Order Synced: $total_orders_synced\n";
+            $msg .= "Total Order Created: $total_orders_created\n";
+            $msg .= "Total Unique Ids Missing on Gateway: $total_orders_missing_on_gateway\n";
+            PrestaShopLogger::addLog($msg, 1, null, $payment_module->name, $payment_module->id, true);
+            exit($msg);
+        } else {
+            $msg = "$this->cron_msg_prefix not run, AltaPay's Payment module not enabled.";
+            PrestaShopLogger::addLog($msg, 3);
+            exit($msg);
+        }
+    }
+
+    /**
+     * Retrieve unique_id values from altapay_transaction
+     * that do not exist in altapay_order.
+     *
+     * @return array|false
+     */
+    private function findMissingAltaPayTransactionRecords($payment_module)
+    {
+        try {
+            // Define the query
+            $query = 'SELECT id, unique_id, id_cart FROM ' . _DB_PREFIX_ . 'altapay_transaction AS t 
+        WHERE is_processed_by_cron = 0 AND is_cancelled != 1 AND NOT EXISTS ( SELECT 1 FROM ' . _DB_PREFIX_ . 'altapay_order AS o WHERE o.unique_id = t.unique_id )
+        LIMIT 10';
+            // Get a reference to the database object
+            $db = Db::getInstance();
+            // Execute the query, fetch & return the results
+            return $db->executeS($query);
+        } catch (Exception $e) {
+            PrestaShopLogger::addLog("Cron error: could not find missing transaction records, DB error: {$e->getMessage()}", 3, null, $payment_module->name, $payment_module->id, true);
+
+            return false;
+        }
+    }
+
+    /**
+     * Get transaction data from AltaPay by shop_orderid
+     *
+     * @param $shop_orderid
+     * @param $payment_module
+     *
+     * @return false|object
+     */
+    private function getTransactionFromAltaPayByShopOrderId($shop_orderid, $payment_module)
+    {
+        try {
+            $api = new API\PHP\Altapay\Api\Others\Payments(getAuth());
+            $api->setShopOrderId($shop_orderid);
+            $paymentDetails = $api->call();
+
+            if (empty($paymentDetails) or $paymentDetails[0]->ReservedAmount == 0) {
+                return false;
+            }
+
+            $fraudConfig = Tools::getValue('enable_fraud', Configuration::get('enable_fraud'));
+            $enableReleaseRefund = Tools::getValue('enable_release_refund', Configuration::get('enable_release_refund'));
+            if ($fraudConfig && $enableReleaseRefund && strtolower($paymentDetails[0]->FraudRecommendation) === 'deny') {
+                PrestaShopLogger::addLog("Cron error: fraud payment shop_orderid: $shop_orderid", 3, null, $payment_module->name, $payment_module->id, true);
+
+                return false;
+            }
+
+            $response['nature'] = $paymentDetails[0]->PaymentNature ?? '';
+            $response['Transactions'] = $paymentDetails;
+
+            return json_decode(json_encode($response));
```

Comment:
I can see you are doing `json_decode(json_encode(` in many places. Why is that?

Reply:
@emicha We're modifying the structure of the object received from payments API to accommodate additional properties, namely `nature` and `Transactions`. These elements are essential for compatibility with the existing function that processes callback responses.

By using `json_decode(json_encode(`, we're ensuring that our response adheres to this structure. This approach intends to provide simple & seamless integration without breaking any existing flow.

Reply:
@maf-embraceit It is not right to take a response from an API, and modify it in order to pass it to another function / class. You should create your own entity, that will be populated with information of your response, and extended with other information you want to add to it. Relying on arrays is also prone to side effects, because you do not have reference of the type, and how fields are used.

I will not ask you to modify now, but we need to move away from this pattern in the future.

Reply:
Got it, will certainly refactor this in future.